onnxruntime/core/providers/cpu/nn/conv_transpose_attributes.h

```diff
@@ -176,6 +176,8 @@ struct ConvTransposeAttributes : public ConvAttributes {
 
   void DistributePadding(AutoPadType pad_type, const int64_t& total_pad,
                          int64_t& pad_head, int64_t& pad_tail) const {
+    // TODO(jcwchen): #9740 ORT 1.13 will correct the logic by switching them to meet ONNX spec
+    LOGS_DEFAULT(WARNING) << "The pad result for SAME_UPPER and SAME_LOWER will be corrected in next ORT 1.13 release.";
```

Comment:
Do you want to only print this warning log ONLY if the auto_pad type is SAME_LOWER/SAME_UPPER ? This function seems to be called with all auto_pad types...

Reply:
Good point. Otherwise, I was making unnecessary spam. It seems that the behavior change here will only infect SAME_UPPER/SAME_LOWER, because VALID/NOTSET shouldn't use the pad value here. Correct the comment below to prevent future confusion.

Reply:
So I am a bit confused after the modification of the comment in the else branch - is the DistributePadding() method called for NOTSET/VALID cases as well or is it not ?

Reply:
Good question -- at least according to ONNX document, NOTSET should not compute anything related to pad_head or pad_tail anyway because in that case explicit padding will be used. However, it's unclear for VALID case in ONNX spec and there is an ONNX issue about this ambiguity: https://github.com/onnx/onnx/issues/3254. I think it would be great if we can solve all issues with one shot. It seems to me that VAILD might means "no padding"? (I saw [old document for Conv](https://github.com/onnx/onnx/blob/6a89ca2e7bb3222684d14d23dbdb1c8cc1bd9ad2/onnx/defs/nn/old.cc#L238) mentioned it) @gramalingam what's your opinion about the VALID case for ConvTranspose?

If my guess is right, perhaps this [line](https://github.com/microsoft/onnxruntime/blob/f4ba199baddbf176369a8fba7ca53a328a361a6c/onnxruntime/core/providers/cpu/nn/conv_transpose_attributes.h#L207) needs to be corrected as well.

Reply:
Sure. But maybe since the intent of this PR is to just add a waring log (and hopefully get it into the 1.12 release), maybe you just want to print the warning log if pad_type is SAME_UPPER/SAME_LOWER and merge this PR without touching anything else ?